### PR TITLE
[LLT-5423] Prevent gateway IP spoofing

### PR DESCRIPTION
**Pull request by tomaszklak**
Compare the address from which we got the gateway advertisement against the IP to which the advertisement is pointing to in its content. If they don't match, fail processing of that advertisement.

This PR also brings the repo to a more up to date state (rust edition, rust version, new clippy warnings, CI actions etc).

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -371,6 +379,7 @@ impl fmt::Display for SearchError {
             SearchError::HyperError(ref e) => write!(f, "Hyper Error: {}", e),
             #[cfg(feature = "aio")]
             SearchError::InvalidUri(ref e) => write!(f, "InvalidUri Error: {}", e),
+            SearchError::SpoofedIp { src_ip, url_ip } => write!(f, "Spoofed IP from {src_ip} spoofed as {url_ip}"),
```
Comment (gytsto) [inline src/errors.rs]: nit: it sounds weird for me, that double spoofed in one sentence. Maybe just leave `Spoofed IP from {src_ip} as {url_ip}`

**Review by gytsto — COMMENTED** [review]
looks good +1

File: src/errors.rs (hunk 1 above)
Comment (tomaszklak) [reply src/errors.rs]: Done

**Review by LukasPukenis — APPROVED** [review]
Looks nice +1

**Pull request merged**